app/apphandlers/embeddedAppHandler.go

```diff
@@ -231,6 +232,23 @@ func (a *embeddedApp) StartRudderCore(ctx context.Context, options *app.Options)
 
 	adaptiveLimit := payload.SetupAdaptiveLimiter(ctx, g)
 
+	var geoEnricher enricher.PipelineEnricher
+	if config.GetBool("GeoEnrichment.enabled", false) {
+
+		dbProvider, err := enricher.NewGeoDBProvider(config, a.log)
+		if err != nil {
+			return fmt.Errorf("creating new instance of db provider: %w", err)
```

Comment:
[minor] May be we can we more specific here?
```suggestion
			return fmt.Errorf("creating new instance of geo db provider: %w", err)
```